src/main/java/com/google/flourbot/entity/EntityInterface.java

```diff
@@ -0,0 +1,8 @@
+package com.google.flourbot.entity;
+
+import java.lang.Exception;
+import com.google.flourbot.datastorage.FirebaseDataStorage;
+
+public interface EntityInterface {
+    public Macro getMacro(String userEmail, String macroName) throws Exception;
```

Comment:
using generic exception is discouraged because you don't know what can be really thrown. Following up on our discussion today. If something bad can happen that can react to (which is the case here) we should return an optional. If there are any checked exceptions that we cannot react to (e.g. broken db connection) we should turn them into checked exceptions. Please read https://www.geeksforgeeks.org/checked-vs-unchecked-exceptions-in-java/ if you would like to learn more.

Reply:
so to be clear, let's return Optional<Macro> here.

Reply:
Thanks for letting me know about Optional! In the new pull request, null pointer/value does not exist Exceptions are turned into Optional.